api/handlers/shared/status.go

```diff
@@ -55,20 +56,24 @@ func Status(c *websocket.Conn) {
 	}
 
 	for {
-		err := k8sClient.Get(context.Background(), stsKey, sts)
-		stsNotFound := apierrors.IsNotFound(err)
 
-		err = k8sClient.Get(context.Background(), key, pod)
+		err := k8sClient.Get(context.Background(), key, pod)
 		if err != nil {
-			if stsNotFound {
+
+			// is the pod error due to sts has been deleted ?
+			stsErr := k8sClient.Get(context.Background(), stsKey, sts)
+			if apierrors.IsNotFound(stsErr) {
 				return
```

Comment:
why did you break the connection with return without handling  the stsErr?

Reply:
I return to close the ws connection. No need to handle the error. There's no std/pod to get the status from.